src/grpc/mod.rs

```diff
@@ -38,6 +57,19 @@ impl ChainFollowerRequestService for AyaCardanoRPCServer {
 
         //select * from block b where b.block_no = (select min(block_no-1) from block b where b.epoch_no = 209) or b.block_no = (select max(block_no) from block b where b.epoch_no = 209) order by block_no DESC;
         let rtype = request.into_inner();
+
+        // Check cache hit
+        let cache_key = encode_prost_message(&rtype);
+        let mut cache = self.short_cache.lock().await;
+        match cache.cache_get(&cache_key) {
+            Some(cached) => {
+                log::debug!("CACHE: Some found {:?}", cached);
+                let output : EventResponse = prost::Message::decode(cached.as_slice()).unwrap();
```

Comment:
We should avoid `unwrap`ing as much as possible because it can cause a runtime panic, also the compiler can optimise return values by reordering or coalescing them it's unable to in the area of code that could panic because of the side effects.